src/data_collector/polygon_fundamentals/client.py

```
@@ -103,62 +140,118 @@ async def _load_from_cache(self, cache_path: Path) -> Optional[Dict[str, Any]]:
+        async with self.session.get(url, params=params) as response:
+            if response.status == 200:
+                try:
+                    data = await response.json()
+                    return data
+                except Exception as e:
+                    raise ValueError(f"Failed to parse JSON response: {e}") from e
+            elif response.status == 429:  # Rate limited
+                error_text = await response.text()
+                logger.warning(f"Rate limited (429): {error_text}")
+                # Let retry framework handle this as a retryable error
+                raise aiohttp.ClientError(f"Rate limited: {error_text}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)